homeassistant/components/notify/rocketchat.py

```diff
@@ -0,0 +1,69 @@
+"""
+Rocket.Chat notification service.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/notify.rocketchat/
+"""
+import asyncio
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_URL, CONF_USERNAME, CONF_PASSWORD)
+from homeassistant.components.notify import (
+    ATTR_DATA, PLATFORM_SCHEMA,
+    BaseNotificationService)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['rocketchat-API==0.6.1']
+
+CONF_ROOM = 'room'
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_URL): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_ROOM): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_get_service(hass, config, discovery_info=None):
+    """Return the notify service."""
+    username = config.get(CONF_USERNAME)
+    password = config.get(CONF_PASSWORD)
+
+    url = config.get(CONF_URL)
+    room = config.get(CONF_ROOM)
+
+    return RocketChatNotificationService(hass, url, username, password, room)
+
+
+class RocketChatNotificationService(BaseNotificationService):
+    """Implement the notification service for Rocket.Chat."""
+
+    def __init__(self, hass, url, username, password, room):
+        """Initialize the service."""
+        from rocketchat_API.rocketchat import RocketChat
+        from rocketchat_API.APIExceptions.RocketExceptions import (
+            RocketConnectionException, RocketAuthenticationException)
+        self._url = url
```

Comment:
The url is only used in this method, so it doesn't need to be an instance attribute.

Reply:
Fixed in 4ab601786.